roxctl/common/printer/tabularprinter_factory.go

```diff
@@ -33,22 +52,27 @@ func NewTabularPrinterFactory(headers []string, rowJSONPathExpression string) *T
 }
 
 // NewTabularPrinterFactoryWithAutoMerge creates new TabularPrinterFactory with the injected default values
-func NewTabularPrinterFactoryWithAutoMerge(headers []string, columnsToMerge []string, rowJSONPathExpression string) *TabularPrinterFactory {
+func NewTabularPrinterFactoryWithAutoMerge() *TabularPrinterFactory {
 	return &TabularPrinterFactory{
 		Merge:                 true,
-		Headers:               headers,
-		RowJSONPathExpression: rowJSONPathExpression,
-		columnsToMerge:        columnsToMerge,
+		Headers:               defaultImageScanHeaders,
+		RowJSONPathExpression: defaultImageScanJSONPathExpression,
+		columnsToMerge:        defaultColumnsToMerge,
 	}
 }
 
 // AddFlags will add all tabular printer specific flags to the cobra.Command
 func (t *TabularPrinterFactory) AddFlags(cmd *cobra.Command) {
 	cmd.PersistentFlags().BoolVar(&t.Merge, "merge-output", t.Merge, "Merge duplicate cells in prettified tabular output")
-	cmd.PersistentFlags().StringSliceVar(&t.Headers, "headers", t.Headers, "Headers to print in tabular output")
+	cmd.PersistentFlags().StringSliceVar(&t.Headers, "headers", defaultImageScanHeaders, "Headers to print in tabular output"+
+		"Will propagate headers to the table unless --row-jsonpath-expressions is also set.")
 	cmd.PersistentFlags().StringVar(&t.RowJSONPathExpression, "row-jsonpath-expressions", t.RowJSONPathExpression,
 		"JSON Path expression to create a row from the JSON object. This leverages gJSON (https://github.com/tidwall/gjson)."+
 			" NOTE: The amount of expressions within the multi-path has to match the amount of provided headers.")
+	err := t.propagateCustomHeaders()
+	if err != nil {
+		cmd.PrintErrf("%v", err)
```

Comment:
Since this will be presented to the user it will be good to test it

Reply:
I have added tests for the different cases that can happen when propagating headers. Any specific one missing?